### DISCUSSION TITLE
ed25519 server certificates problems with libressl

### DISCUSSION THREAD

**Discussion by bagder** [curl/curl-for-win#78] [General]
Someone tripped over the fact that server signatures signed with ED25519 does not work with curl+libressl (nor with schannel), but does with curl using some of the other backends (for example OpenSSL).

The request for ed25519 server certificate support in libressl remains open: https://github.com/libressl/portable/issues/821

**Comment by vszakats** [comment]
This should probably become a list at one point, other missing features:
- LibreSSL (as of 4.1.0) does not support TLSv1.3 session tickets
  https://github.com/libressl/portable/issues/719
  https://github.com/curl/curl/issues/18031#issuecomment-3144406973
- `SSL_set0_wbio()`: https://github.com/libressl/portable/issues/838
- ECH: https://github.com/libressl/portable/issues/546
- `SSLKEYLOGFILE`:
  https://github.com/curl/curl/issues/13672
  https://marc.info/?l=libressl&m=158908819814107
  https://superuser.com/questions/1871314/curl-for-windows-sslkeylog-file-always-empty-when-running-from-powershell

**Reply by bagder** [reply]
- SSLKEYLOGFILE support too, right?

**Reply by vszakats** [reply]
Correct, added it to the list. It may have been a choice for security IIRC.

**Reply by bagder** [reply]
Right, but for us the reason for the non-support is of course less interesting.

We should probably put this list somewhere that people can find.

**Reply by vszakats** [reply]
Almost forgot, that I mainain a list of caveats for each dependency, here it is for LibreSSL:
https://github.com/curl/curl-for-win/blob/966137499eadcd7658fd754f6c8d176c9db81208/libressl.sh#L6-L26

Not overly visible and not "README" quality though.
